### PR TITLE
Fixes #10355 - Prevent turbolinks loading puppetclass import 

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: app/services/puppet_class_importer.rb [hunk 1]
```diff
@@ -142,7 +142,7 @@ def old_environments
   def db_classes(environment)
     return @foreman_classes[environment] if @foreman_classes[environment]
     return [] unless (env = Environment.find_by_name(environment))
-    @foreman_classes[environment] = env.puppetclasses.includes(:lookup_keys, :class_params)
+    @foreman_classes[environment] = env.puppetclasses
```
Comment (ohadlevy) [inline app/services/puppet_class_importer.rb]: why?

File: app/helpers/puppetclasses_and_environments_helper.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ def import_proxy_select(hash)
 
   def import_proxy_links(hash, classes = nil)
     SmartProxy.with_features("Puppet").map do |proxy|
-      display_link_if_authorized(_("Import from %s") % proxy.name, hash.merge(:proxy => proxy), :class=>classes)
+      display_link_if_authorized(_("Import from %s") % proxy.name, hash.merge(:proxy => proxy), {:class=>classes, :"data-no-turbolink" => true})
```
Comment (ohadlevy) [inline app/helpers/puppetclasses_and_environments_helper.rb:20]: i wonder if it make sense to have a simple modal / ajax action, as the initial page load can be very long if you have a largish set of puppetclasses

File: app/services/puppet_class_importer.rb (hunk 1 above)
Comment (tbrisker) [inline app/services/puppet_class_importer.rb]: Looks like this isn't being used anywhere (bullet is complaining and I couldn't find anywhere it is used).

File: app/helpers/puppetclasses_and_environments_helper.rb (hunk 2 above)
Comment (tbrisker) [inline app/helpers/puppetclasses_and_environments_helper.rb:20]: I think that might make sense; however, is there any way we can speed up the page load? It currently fires several hundred queries at the database, as well as querying the smart proxy - both of them candidates for improvements.

Comment (ohadlevy) [inline app/helpers/puppetclasses_and_environments_helper.rb:20]: IMHO that would be more complex then a simple ajax call to start with, we already added a caching layer on the proxy, and there are two longish operations in the background
1. ask the proxy to scan for puppet manifests file, the proxy already caches the results, but we could consider adding a new api for all environments vs per environment api.
2. compare the current state in the db (env -> classes -> parameters) to the new state. also takes time.
the first action is what slows down the initial page load, and the second action is what slows down after submitting... both can be very long.

**Comment by domcleal** [comment]
Merged as df8887a, thanks @tbrisker.  I'd suggest filing a new ticket about ideas to refactor this page entirely.

**Pull request closed without merging**